pkg/backend/httpstate/backend.go

```diff
@@ -1419,6 +1421,33 @@ func (b *cloudBackend) apply(
 		return nil, nil, err
 	}
 
+	// Note: ShowCopilotSummary can only be set to true via the update cmd (e.g. `pulumi up`)
+	// This code is here so we can capture errors from previews-of-updates as well as updates.
+	// The createAndStartUpdate call above can also disable ShowCopilotSummary if its not enabled in the user's org.
+	if op.Opts.Display.ShowCopilotSummary {
+		eventsChannel := make(chan engine.Event)
+		originalEvents := events
+		events = eventsChannel
+
+		var renderEvents []engine.Event
+		go func() {
+			for e := range eventsChannel {
+				if originalEvents != nil {
+					originalEvents <- e
+				}
+				if e.Internal() {
+					continue
+				}
+				renderEvents = append(renderEvents, e)
+			}
+		}()
+
+		defer func() {
+			close(eventsChannel)
+			b.renderAndSummarizeOuput(ctx, kind, stack, op, renderEvents)
```

Comment:
Probably need a `<-done` chan here.

Reply:
Yea I was wondering -- is this close safe? We know that the call prior to this at the function end won't return until the channel is done with?

Reply:
Yeah, I was following along w/ `PreviewThenPrompt` but that one's gonna have a few seconds before user interactively asks for a diff etc. Will add something here.

Reply:
Or wait, did you mean is the close itself as the called fn might do a write after we close? I assume this being in the defer we're rosy. But maybe some edge cases here?